### PR TITLE
SONAR-7741 upgrade OK HTTP to 3.3.1

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: sonar-ws/src/main/java/org/sonarqube/ws/client/HttpConnector.java [hunk 1]
```diff
@@ -78,27 +78,33 @@ private HttpConnector(Builder builder, JavaVersion javaVersion) {
...
 
     ConnectionSpec tls = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
       .allEnabledTlsVersions()
       .allEnabledCipherSuites()
       .supportsTlsExtensions(true)
       .build();
-    this.okHttpClient.setConnectionSpecs(asList(tls, ConnectionSpec.CLEARTEXT));
-    this.okHttpClient.setSslSocketFactory(createSslSocketFactory(javaVersion));
+    okHttpClientBuilder.connectionSpecs(asList(tls, ConnectionSpec.CLEARTEXT));
+    okHttpClientBuilder.sslSocketFactory(createSslSocketFactory(javaVersion));
```
Comment (SonarTech) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/HttpConnector.java]: MINOR Remove this use of "sslSocketFactory"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: sonar-ws/src/main/java/org/sonarqube/ws/client/HttpConnector.java (hunk 1 above)
Comment (sns-seb) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/HttpConnector.java]: @simonbrandhof if you have insight on how to use the non-deprecated version of this method, I'm all ears

**Comment by henryju** [comment]
enableTls12InJava7 is useless but apart from that LGTM

**Pull request merged**